config/jobs/image-pushing/k8s-staging-secrets-store-sync.yaml

```diff
@@ -0,0 +1,34 @@
+postsubmits:
+  # This is the github repo we'll build from. This block needs to be repeated
+  # for each repo.
+  kubernetes-sigs/secrets-store-sync-controller:
+    # The name should be changed to match the repo name above
+    - name: secrets-store-sync-controller-push-image
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        # This is the name of some testgrid dashboard to report to.
+        # If this is the first one for your sig, you may need to create one
+        testgrid-dashboards: sig-auth-secrets-store-sync-controller, sig-k8s-infra-gcb
+      decorate: true
+      # we only need to run if necessary (e.g.: the version was bumped)
+      run_if_changed: 'docker/Makefile'
+      # this causes the job to only run on the master branch. Remove it if your
+      # job makes sense on every branch (unless it's setting a `latest` tag it
+      # probably does).
+      branches:
+        - ^release-.*
```

Comment:
/hold

Are you sure you only want to build the image on release branches? This job usually runs on every merge to master branch

Reply:
We only want this job to run in release branches when we cut a release. The job is configured to only run when `docker/Makefile` is changed which we update at the release time. 

This job is a replica of the other job https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-csi-secrets-store.yaml we have for Secrets Store CSI Driver.

Reply:
Alright
/hold cancel